src/languages/lisp.md

```diff
@@ -0,0 +1,168 @@
+
+## QuickLisp options
+
+If you wish to change the distributions that QuickLisp is using, you can specify those as follows, specifying a distribution name, its URL and, an optional version:
+
+```yaml
+runtime:
+    quicklisp:
+        <distribution name>:
+            url: "..."
+            version: "..."
+```
+
+For example:
+
+```yaml
+runtime:
+    quicklisp:
+        quicklisp:
+            url: 'http://beta.quicklisp.org/dist/quicklisp.txt'
+            version: '2019-07-11'
+```
+
+
+## Platform.sh variables
+
+Platform.sh exposes relationships and other configuration as [environment variables](/development/variables.md). Most notably, it allows a program to determine at runtime what HTTP port it should listen on and what the credentials are to access [other services](/configuration/services.md).
+
+To get the `PORT` environment variable (the port on which your web application is supposed to listen) you would:
+
+```lisp
+(parse-integer (uiop:getenv "PORT"))
+```
+
+
+## Building and running the application
+
+Assuming `example.lisp` and `example.asd` are present in your repository, the application will be automatically built on push.  You can then start it from the `web.commands.start` directive.  Note that the start command _must_ run in the foreground. Should the program terminate for any reason it will be automatically restarted. In the example below we sleep for a very, very long time. You could also choose to join the thread of your web server, or use other methods to make sure the program does not terminate.
+
+The following basic `.platform.app.yaml` file is sufficient to run most Lisp applications.
+
+```yaml
+name: app
+type: lisp:1.5
+web:
+    commands:
+        start: ./example
+    locations:
+        /:
+            allow: false
+            passthru: true
+disk: 512
+```
+
+Note that there will still be a proxy server in front of your application.  If desired, certain paths may be served directly by our router without hitting your application (for static files, primarily) or you may route all requests to the Lisp application unconditionally, as in the example above.
+
+# Accessing Services
+
+The services configuration is available in the environment variable `PLATFORM_RELATIONSHIPS`. 
+
+To parse them, add the dependencies to your `.asd` file:
+
+```lisp
+:depends-on (:jsown :babel :s-base64)
+````
+
+The following is an example of accessing a PostgreSQL instance:
+
+```lisp
+(defun relationships ()
+  (jsown:parse
+   (babel:octets-to-string
+    (with-input-from-string (in (uiop:getenv "PLATFORM_RELATIONSHIPS"))
+      (s-base64:decode-base64-bytes in)))))
+```
+
+Given a relationship defined in `.platform.app.yaml`:
+
+```yaml
+relationships:
+  pg: postgresql:postgresql
+```
+
+The following would access that relationship, and provide your Lisp program the credentials to connect to a PostgreSQL instance. Add this to your `.asd` file:
+
+```lisp
+:depends-on (:postmodern)
+````
```

Comment:
```suggestion
```
```

Reply:
Same as above. Don't accept the suggestion (unfortunately) - just remove one back-tick from the end of the code block.